Started componentizing and refactors some code

Nick-Gabe opened the pull request:
- Refactors victory check. instead of a lot of ifs verifying each possible condition it performs a smart check through the states:
> ```ts
>  const checkVictory = () => {
>    const options = { scissors, rocks, papers };
>    const selected = options[guess];
>    const otherTwo = Object.entries(options).filter(([key]) => key !== guess);
>
>    if (selected.length === 0) {
>      alert("You lose!");
>      setPlaying(false);
>    } else if (otherTwo.every(([key, value]) => value.length === 0)) {
>      alert("You win!");
>      setPlaying(false);
>    }
>  };
> ```
> if the adversaries are 0, you win, if you're 0, you lose. It's way smaller and scalates well since to add a new option you only need to add a new word.

- Started componentizing some elements.
> Instead of creating them directly inside the App, I created separate files for some of them (the bottom part), and grouped inside `BottomContext/index.tsx`. So in app the code amount is way smaller:
> ```tsx
>      <BottomContent
>        guess={guess}
>        setGuess={setGuess}
>        setPlaying={setPlaying}
>      />
> ```
> This way its more organized and reduces the code inside the main file. The ideal would be to break the App file even more, for example generating the papers, scissors and rocks inside a specific file.

- Added some css to make the bottom bar more detailed:
> Default:
> ![image](https://user-images.githubusercontent.com/42651514/197885139-0593b3df-3aeb-4633-9f38-adc4dc5f5a9a.png)
> Hovering:
> ![image](https://user-images.githubusercontent.com/42651514/197885255-52315c42-4176-4c25-a369-e4efe8d5f86a.png)

Nick-Gabe committed:
started componentizing bottom of page

Nick-Gabe committed:
refactors victory check and folder structure

datsfilipe reviewed (commented):
I like the way you did the refactor, so I'm planning to stick with that way. The problem is just that you missed some points:
- it's intent is not  to be scalable, since it's just a rock, paper, scissors game (but it's cool that it is now 👍);
- and if we just count victory or lose when the last guesses option/or the otherTwo are gone, the user might have to wait too much for a known already result.

Commit:
fix: give the result before the game ends

Sometimes it's possible to know the result before the end of the game, so we let the user know the results.
